### ISSUE TITLE
Test failure: Smoke_Tests

### ISSUE THREAD

**Issue by aspnet-hello** [aspnet/MusicStore#869]
This test [fails](http://aspnetci/viewLog.html?buildId=558373&buildTypeId=XPlat_Windows_Win2008r2_Universe) occasionally with the following error:
```
System.OperationCanceledException : Failed to connect, retry canceled.
   at Microsoft.AspNetCore.Server.IntegrationTesting.RetryHelper.RetryRequest(Func`1 retryBlock, ILogger logger, CancellationToken cancellationToken, Int32 retryCount)
   at E2ETests.SmokeTests.RunTestsAsync(DeploymentResult deploymentResult, ILogger logger) in C:\BuildAgent\work\33bdfc1cae7b2a38\modules\MusicStore\test\MusicStore.E2ETests\SmokeTests.cs:line 159
   at E2ETests.SmokeTests.Smoke_Tests(TestVariant variant) in C:\BuildAgent\work\33bdfc1cae7b2a38\modules\MusicStore\test\MusicStore.E2ETests\SmokeTests.cs:line 69
--- End of stack trace from previous location where exception was thrown ---
------- Stdout: -------
| [0.001s] TestLifetime Information: Starting test Smoke_Tests-Server: IISExpress, TFM: netcoreapp2.2, Type: Standalone, Arch: x64, ANCM: V2, Host: InProcess at 2018-09-21T18:43:28
| [0.001s] TestLifetime Information: Starting test SmokeTestSuite_Server: IISExpress, TFM: netcoreapp2.2, Type: Standalone, Arch: x64, ANCM: V2, Host: InProcess at 2018-09-21T18:43:28
| [0.002s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: Deploying [Variation] :: ServerType=IISExpress, Runtime=CoreClr, Arch=x64, BaseUrlHint=, Publish=False
| [0.002s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: Executing command dotnet publish  --output "C:\BuildAgent\temp\buildTmp\c9033c0a0b76465297ed173f3aabb2da" --framework netcoreapp2.2 --configuration Release --no-restore -p:VerifyMatchingImplicitPackageVersion=false --runtime win7-x64 
| [0.234s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: dotnet-publish stdout: Microsoft (R) Build Engine version 15.8.166+gd4e8d81a88 for .NET Core
| [0.234s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: dotnet-publish stdout: Copyright (C) Microsoft Corporation. All rights reserved.
| [1.012s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: dotnet-publish stdout:   MusicStore -> C:\BuildAgent\work\33bdfc1cae7b2a38\modules\MusicStore\samples\MusicStore\bin\Release\netcoreapp2.2\win7-x64\MusicStore.dll
| [1.013s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: dotnet-publish stdout:   MusicStore -> C:\BuildAgent\work\33bdfc1cae7b2a38\modules\MusicStore\samples\MusicStore\bin\Release\netcoreapp2.2\win7-x64\MusicStore.Views.dll
| [2.266s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: dotnet-publish stdout:   MusicStore -> C:\BuildAgent\temp\buildTmp\c9033c0a0b76465297ed173f3aabb2da\
| [2.316s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: dotnet publish finished with exit code : 0
| [2.318s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: Attempting to start IIS Express on port: 60628
| [2.320s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: Executing command : C:\Program Files\IIS Express\iisexpress.exe /site:HttpTestSite /config:C:\BuildAgent\temp\buildTmp\tmp85E6.tmp /trace:error /systray:false
| [2.320s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: SET ASPNETCORE_ENVIRONMENT=SocialTesting
| [2.321s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: SET ASPNETCORE_DETAILEDERRORS=true
| [2.321s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: SET Data__DefaultConnection__ConnectionString=Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=MusicStore_Test_9c9d60ebb632496eaf97bb93c79fad40;Integrated Security=True;MultipleActiveResultSets=True;Connect Timeout=30
| [2.327s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress Process 764 started
| [2.376s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: Starting IIS Express ...
| [2.390s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: Initializing the W3 Server Started CTC = 805864937 
| [2.394s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: W3 Server initializing WinSock.  CTC = 805864937 
| [2.394s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: W3 Server WinSock initialized.  CTC = 805864937 
| [2.394s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: W3 Server ThreadPool initialized (ipm has signalled).  CTC = 805864937 
| [2.412s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: Start listenerChannel http:0
| [2.413s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: Successfully registered URL "http://localhost:60628/" for site "HttpTestSite" application "/"
| [2.414s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: Registration completed for site "HttpTestSite"
| [2.414s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: AppPool 'IISExpressAppPool' initialized
| [2.414s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: InitComplete event signalled
| [2.414s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: IIS Express is running.
| [2.414s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: Started iisexpress successfully. Process Id : 764, Port: 60628
| [2.414s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: Application ready at URL: http://localhost:60628/
| [2.41
...
```
Other tests within that build may have failed with a similar message, but they are not listed here. Check the link above for more info.

This test failed on master.

CC @mkArtakMSFT

This issue was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
Please use this workflow to address this flaky test issue, including checking applicable checkboxes and filling in the applicable "TODO" entries:

* Is this actually a flaky test?
  * No, this is a regular test failure, fix the test/product (TODO: Link to commit/PR)
  * Yes, proceed below...

* Is this test failure caused by product code flakiness? (Either this product, or another product this test depends on.)
  * [ ] File a bug against the product (TODO: Link to other bug)
  * Is it possible to change the test to avoid the flakiness?
    * Yes? Go to the "Change the test!" section.
    * No?
      * [ ] Disable the test (TODO: Link to PR/commit)
      * [ ] Wait for other bug to be resolved
      * [ ] Wait for us to get build that has the fix
      * [ ] Re-enable our test (TODO: Link to PR/commit)
      * [ ] Close this bug

* Is it that the test itself is flaky? This includes external transient problems (e.g. remote server problems, file system race condition, etc.)
  * Is there is a way to change our test to avoid this flakiness?
    * Yes? Change the test!
      * [ ] Change the test to avoid the flakiness, for example by using a different test strategy, or by adding retries w/ timeouts (TODO: Link to PR/commit)
      * [ ] Run the test 100 times locally as a sanity check.
      * [ ] Close this bug
    * No?
      * Is there any logging or extra information that we could add to make this more diagnosable when it happens again?
        * Yes?
            * [ ] Add the logging (TODO: Link to PR/commit)
        * No?
            * [ ] Delete the test because flaky tests are not useful (TODO: Link to PR/commit)

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
Smoke_Tests [failed with about the same error](http://aspnetci/viewLog.html?buildId=558373&buildTypeId=XPlat_Windows_Win2008r2_Universe) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by mkArtakMSFT** [comment]
@dougbu, can you please look into this? Thanks!

**Comment by dougbu** [comment]
Looks like a dependency issue involving System.IO.Pipelines, System.Memory and System.Runtime.CompilerServices.Unsafe. Will update dependencies and try to reproduce locally, then examine the dependency graph&hellip;

/cc @halter73 @pakrym

**Comment by aspnet-hello** [comment]
There were 6 failures [with about the same error](http://aspnetci/viewLog.html?buildId=558745&buildTypeId=Lite_UniverseTest) on master:
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by dougbu** [comment]
Likely related to aspnet/Home#3544 (thanx for the pointer @ryanbrandenburg!)

**Comment by aspnet-hello** [comment]
There were 6 failures [with about the same error](http://aspnetci/viewLog.html?buildId=559368&buildTypeId=XPlat_Windows_Win8_Universe) on master:
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by dougbu** [comment]
Note: This is not about _flaky_ tests. The tests are obviously broken at the moment -- they haven't succeeded since about the 18th where they're enabled e.g. Win10-Universe.

**Comment by dougbu** [comment]
Searched a diagnostic build log and found no version conflicts for System.Runtime.CompilerServices.Unsafe, System.Memory or System.IO.Pipelines.

Resolved versions are as expected:
- System.Runtime.CompilerServices.Unsafe: 4.6.0-preview1-26907-04
- System.Memory: 4.5.0
- System.IO.Pipelines: 4.6.0-preview1-26907-04

Could not get the MusicStore site to file; problems appear specific to tests. Switching to aspnet/Home#3544 in hopes that makes investigation easier.

**Comment by aspnet-hello** [comment]
There were 6 failures [with about the same error](http://aspnetci/viewLog.html?buildId=560048&buildTypeId=XPlat_Windows_Win8_Universe) on master:
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
There were 6 failures [with about the same error](http://aspnetci/viewLog.html?buildId=560221&buildTypeId=XPlat_Windows_Win7_Universe) on master:
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
There were 6 failures [with about the same error](http://aspnetci/viewLog.html?buildId=561382&buildTypeId=Lite_UniverseTest) on master:
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests
- Smoke_Tests


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
@dougbu was this resolved by @natemcmaster's work?

**Comment by dougbu** [comment]
@ryanbrandenburg what work in particular?

**Comment by ryanbrandenburg** [comment]
aspnet/Home#3544.

**Comment by dougbu** [comment]
@ryanbrandenburg, now that we've seen a few days without these failures, looks like it.

FYI for others, workaround was aspnet/Universe@91ad43b5f9 and we're tracking a real fix for dotnet/corefx#32457 in aspnet/Universe#1421
